src/types.d.ts

```diff
@@ -0,0 +1,18 @@
+export interface EventEmitterFactory {
```

Comment:
Is defining types for `EventEmitter` still necessary?  I just checked the branch out, removed this file and all the imports referencing it and could build types without errors.

Reply:
Good catch! Removing it